File: 7785.py
import sys

n = int(sys.stdin.readline())

# 입출입 명단 디셔너리 선언
name_dict = {}

# 명단 업데이트
for _ in range(n):
    name, state = map(str, sys.stdin.readline().split())

    if name in name_dict:
        name_dict[name] = state
    else:
        name_dict[name] = state

#들어와 있는 사람 리스트 선언
enter_list = []

# 명단에서 입장인 사람 리스트에 추가 
for name in name_dict:
    if name_dict[name] == "enter":
        enter_list.append(name)

# 사전 역순 정렬
enter_list.sort(reverse=True)

for i in enter_list:
    print(i)

File: _7785.java
import java.io.*;
import java.util.*;

public class _7785 {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());

        Map<String, Integer> map = new HashMap<>();
        String name, state;

        for(int i=0; i<n; i++){
            StringTokenizer st = new StringTokenizer(br.readLine());
            name = st.nextToken();
            state = st.nextToken();

            if(state.equals("enter")){
                map.put(name, null);
            }
            else{
                map.remove(name);
            }
        }

        ArrayList<String> arr = new ArrayList<>(map.keySet());

        Collections.sort(arr, Collections.reverseOrder());

        StringBuilder sb = new StringBuilder();

        for(String temp: arr){
            sb.append(temp).append("\n");
        }
        System.out.print(sb);
    }
}
